src/Microsoft.DotNet.Build.Tasks/PackageFiles/init-tools.cmd

```diff
@@ -56,6 +56,24 @@ if not [%NET46_PUBLISH_ERROR_LEVEL%]==[0] (
 	exit /b %NET46_PUBLISH_ERROR_LEVEL%
 )
 
+set ILASM_PROJECTJSON=%BUILDTOOLS_PACKAGE_DIR%\ilasm\project.json
```

Comment:
Yes, it's because ilasm links against CoreCLR, and against a newer version than we use for the rest of the tools. Publishing ilasm alongside the rest of the tools would not work because it has conflicting dependencies, which will break the rest of the tools.